### ISSUE TITLE
CNTK outputs the same results regardless of input value

### ISSUE THREAD

**Issue by jkhlot** [Microsoft/CNTK#2403]
When I use CNTK to build DL models for regression, the model always outputs the same results regardless of input value, the following has been tried:
1) normalize the input value
2) add BatchNorm layer in the model structure

But it dose not work, any suggestions to handle this issue?

**Comment by cha-zhang** [comment]
Most like your model is never successfully trained. How is your train loss look like?

**Comment by jkhlot** [comment]
@cha-zhang Hi Mr Zhang, I define my loss function : 
flag = greater( element_times(model_output, true_label), 0)
cost_fun = reduce_sum( element_select(flag, element_min( abs(model_output), abs(true_label) ),  -plus( abs(model_output),abs(true_label) )  )  )

you can regard the  model_output as the predicted stock_price , true_label as the real stock_price, the cost_fun is to minimize the error between them

**Comment by cha-zhang** [comment]
I don't mean the loss function itself, but rather how your loss change during training. Is there any indication of loss going lower over time?

**Comment by jkhlot** [comment]
@cha-zhang The loss gradually drop when increasing the epoch with some fluctuation, but the output value kept unchanged when training with different input values.
I have tried to lower the learning rate, but it did not work

**Comment by cha-zhang** [comment]
Do you preprocess your data? If you do, do you process the data the same way as when you do training?

**Comment by jkhlot** [comment]
@cha-zhang Yes, all the input data(images) are normalized. I have observed the training and testing data prediction result, both of them predict the same value regardless of the input.
Lower the learning rate helps in the first few epoches, but useless in the following epoches..

btw, I wonder it's the issue of my cost function, because the element_min function is not convex , the optimizer may not find a global optimized value, how do you think about it ?
